Classes/Domain/Model/Changes/MoveAfter.php

```diff
@@ -43,7 +43,7 @@ public function apply()
             $before = self::cloneNodeWithNodeData($this->getSubject());
             $parent = $before->getParent();
 
-            $nodeName = $this->generateUniqueNodeName($this->getSiblingNode()->getParent());
+            $nodeName = $this->generateUniqueNodeNameIfNeeded($this->getSiblingNode()->getParent(), $this->getSubject());
```

Comment:
What about moving the logic here to make this a little more explicit? So along the lines of:
```
if ($this->requiresNewNodeName(...)) {
  $nodeName = $this->generateUniqueNodeName($this->getSiblingNode()->getParent());
  $this->getSubject()->moveAfter($this->getSiblingNode(), $nodeName);
} else {
  $this->getSubject()->moveAfter($this->getSiblingNode());
}
```

Reply:
I like!

Reply:
Sure I can do that if you wish, I see the benefits of the explicit style.
But this needs to be done three times and then it's just duplicated code.